### PR TITLE
update solution of #42 [faster, shorter, more intuitive]

### PR DESCRIPTION
**Pull request by ab12gu**
[//]: # "Pull Request Template"
[//]: # "Replace the placeholder values in the template below"

- **File(s) Modified**: 0042-trapping-rain-water.py
- **Language(s) Used**: python
- **Submission URL**: https://leetcode.com/problems/trapping-rain-water/submissions/1034548101/

### CODE REVIEW

**Review by felivalencia3 — APPROVED** [review]

**Comment by Shubashree** [comment]
@ab12gu Can you please provide a video/textual explanation of the above logic? 

It will be useful to draw parallels between https://www.youtube.com/watch?v=ZI2z5pq0TqA and the updated code.

File: python/0042-trapping-rain-water.py [hunk 1]
```diff
@@ -1,18 +1,16 @@
 class Solution:
     def trap(self, height: List[int]) -> int:
-        if not height:
-            return 0
```
Comment (kmanikumar) [inline python/0042-trapping-rain-water.py:3]: can we get some comments I have no idea what's going on here? :) thanks

**Comment by neetcode-gh** [comment]
Hi, I know that this commit simplifies the code, but users prefer the code to be the same as the video. Please don't update existing python solutions unless there is a good reason to do so. Feel free to tag me if needed.

@ab12gu

**Comment by ab12gu** [comment]
> Hi, I know that this commit simplifies the code, but users prefer the code to be the same as the video. Please don't update existing python solutions unless there is a good reason to do so. Feel free to tag me if needed.
> 
> 
> 
> @ab12gu 

Hi,

Actually it was only the first user. The second person Kasi, is a friend that was making fun of me/commentor btw. 

In regards, I don't think the first commenter will lead to a general consensus.

I can add a comment update on how it they are related to the video this week otherwise.

Let me know your thoughts.


@neetcode-gh

**Pull request merged**